services/user/CommonSys/common/packages/wasm-loader/src/importables.js

```diff
@@ -1,21 +1,164 @@
-export async function funccall(params) {
-  return "funcCall";
+const UNINITIALIZED = "uninitialized";
+
+let callerService = UNINITIALIZED;
+let callerOrigin = UNINITIALIZED;
+let myService = UNINITIALIZED;
+let myOrigin = UNINITIALIZED;
+
+function pluginId(namespace, pkg) {
+    return {
+        service: namespace,
+        plugin: pkg
+    };
 }
 
-export function getLoggedInUser() {
-  return "alice";
+function commonErr(val) {
+    return {
+        code: 0,
+        producer: pluginId("common", "plugin"),
+        message: val
+    };
 }
 
-export async function prnt(string) {
-  console.log(`from imported code:`, string);
+function send(req) {
+    try {
+        const xhr = new XMLHttpRequest();
+        xhr.open(req.method.toString(), req.uri, false);
+        const requestHeaders = new Headers(req.headers);
+        for (let [name, value] of requestHeaders.entries()) {
+            if (name !== "user-agent" && name !== "host") {
+                xhr.setRequestHeader(name, value);
+            }
+        }
+        xhr.send(req.body && req.body.length > 0 ? req.body : null);
+        if (xhr.status === 500) {
+            throw commonErr(`${xhr.response}`);
+        }
+        const body = xhr.response || undefined;
+        const headers = [];
+        xhr.getAllResponseHeaders()
+            .trim()
+            .split(/[\r\n]+/)
+            .forEach((line) => {
+                var parts = line.split(": ");
+                var key = parts.shift();
+                var value = parts.join(": ");
+                headers.push([key, value]);
+            });
+        return {
+            status: xhr.status,
+            headers,
+            body
+        };
+    } catch (err) {
+        throw new Error(err.message);
+    }
```

Comment:
Any reason not to leverage something like the fetch api? Doing AJAX like this is *old school* :)
const resp = fetch(req.uri, {
  method: req.method.toString(),
  headers: new Headers(req.headers)
})

Reply:
Yes, the reason is that this magic allows it to be a synchronous import with no groundhog day!